funcx_sdk/funcx/sdk/executor.py

```diff
@@ -83,346 +102,433 @@ def __repr__(self):
 class FuncXExecutor(concurrent.futures.Executor):
     """
     The ``FuncXExecutor`` class, a subclass of `concurrent.futures.Executor`_, is the
-    preferred approach to collecting results from the funcX Web Service.  Over
+    preferred approach to collecting results from the funcX web services.  Over
     polling (the historical approach) where the web service must be repeatedly
     queried for the status of tasks and results eventually collected in bulk, the
-    ``FuncXExecutor`` class instantiates a WebSocket connection that streams results
+    ``FuncXExecutor`` class instantiates an AMQPS connection that streams results
     directly -- and immediately -- as they arrive at the server.  This is a far more
     efficient paradigm, simultaneously in terms of bytes over the wire, time spent
     waiting for results, and boilerplate code to check for results.
 
     An interaction might look like::
 
         from funcx import FuncXExecutor
-        from funcx.sdk.executor import FuncXFuture
-
-        fxexec = FuncXExecutor()
-        ep_id = "<YOUR_ENDPOINT_UUID>"
 
-        def example_funcx_kernel(num):
+        def task_func(num):
             import time
             time.sleep(num * random.random())  # simulate some processing
             return f"result, from task: {num}"
 
-        futs: list[FuncXFuture] = [
-            fxexec.submit(example_funcx_kernel, task_i, endpoint_id=ep_id)
-            for task_i in range(1, 21)
-        ]
-        # FuncXFuture is a subclass of concurrent.futures.Future
+        ep_id = "<YOUR_ENDPOINT_UUID>"
+        num_tasks = 10
 
-        results, exceptions = [], []
-        for f in concurrent.futures.as_completed(futs, timeout=30):
-            # wait no more than 30s for all results
-            try:
-                results.append(f.result())
-            except Exception as exc:
-                exceptions.append((f.task_id, exc))
+        fxe = FuncXExecutor(endpoint_id=ep_id)
```

Comment:
Good observation.  Was ... trying to avoid that work.  Have done that (as per our discussion earlier today).  Please have a fresh review of that output.